rust_snuba/src/processors/generic_metrics.rs

```diff
@@ -193,6 +194,9 @@ impl Parse for CountersRawRow {
         let timeseries_id =
             generate_timeseries_id(from.org_id, from.project_id, from.metric_id, &from.tags);
         let (tag_keys, tag_values): (Vec<_>, Vec<_>) = from.tags.into_iter().unzip();
+
+        metric!(Gauge: "process_messages.messages.tags_len", tag_keys.len(), "metric_type" => "counter");
```

Comment:
you should be using a timer for all of those tags_len metrics IMO

Reply:
fixed